client/my-sites/upgrades/domain-management/email-forwarding/email-forwarding-add-new.jsx

```diff
@@ -48,31 +59,39 @@ const EmailForwardingAddNew = React.createClass( {
 	onAddEmailForward( event ) {
 		event.preventDefault();
 
-		if ( this.state.formSubmitting ) {
-			return false;
+		if ( ! this.state.showForm ) {
+			this.setState( { showForm: true } );
+			return;
 		}
 
-		this.setState( { formSubmitting: true } );
+		this.formStateController.handleSubmit( ( hasErrors ) => {
+			if ( hasErrors ) {
+				return;
+			}
 
-		upgradesActions.addEmailForwarding( this.props.selectedDomainName, this.state.mailbox, this.state.destination, ( error ) => {
-			this.setState( { formSubmitting: false } );
+			const fieldValues = formState.getAllFieldValues( this.state.fields );
```

Comment:
This is a good use case for destructuring: you can write this assignment as
```
const { email, mailbox } = formState.getAllFieldValues( this.state.fields );
```
That way, it's immediately known which fields you'll be using (and it's shorter ;)).